blues/ncmc.py

```diff
@@ -1,580 +1,332 @@
+        verbose=False"""
+
+        #Structure of entire system
+        self.structure = structure
+        #Atom indicies for model
+        self.atom_indices = model.atom_indices
+
+        self.system = None
+        self.alch_system = None
+        self.md = None
+        self.alch  = None
+        self.nc  = None
+
+        self.opt = opt
+    def generateAlchSystem(self, system, atom_indices):
+        """Returns the OpenMM System for alchemical perturbations.
+
+        Parameters
+        ----------
+        system : openmm.System
+            The OpenMM System object corresponding to the reference system.
+        atom_indices : list
+            Atom indicies of the model.
         """
-        if residueList == None:
-            residueList = self.residueList
-        self.md_simulation = md_simulation
-        self.dummy_simulation = dummy_simulation
-        self.nc_context = nc_context
-        self.nc_integrator = nc_integrator
-        self.get_particle_masses(self.md_simulation.system)
-        if print_output == sys.stdout:
-            print_file = sys.stdout
+        factory = alchemy.AlchemicalFactory()
+        alch_region = alchemy.AlchemicalRegion(alchemical_atoms=atom_indices)
+        alch_system = factory.create_alchemical_system(system, alch_region)
+        return alch_system
+
+    def generateSystem(self, structure, nonbondedMethod='PME', nonbondedCutoff=10,
+                       constraints='HBonds', **opt):
+        """Returns the OpenMM System for the reference system.
+
+        Parameters
+        ----------
+        structure: parmed.Structure
+            ParmEd Structure object of the entire system to be simulated.
+        opt : optional parameters (i.e. cutoffs/constraints)
+        """
+        system = structure.createSystem(nonbondedMethod=eval("app.%s" % nonbondedMethod),
+                            nonbondedCutoff=nonbondedCutoff*unit.angstroms,
+                            constraints=None)
+        return system
+
+    def generateSimFromStruct(self, structure, system, nIter, nstepsNC, nstepsMD,
+                             temperature=300, dt=0.002, friction=1,
+                             reporter_interval=1000,
+                             ncmc=False, platform=None,
+                             verbose=False, printfile=sys.stdout,  **opt):
+        """Used to generate the OpenMM Simulation objects given a ParmEd Structure.
+
+        Parameters
+        ----------
+        structure: parmed.Structure
+            ParmEd Structure object of the entire system to be simulated.
+        system :
+        opt : optional parameters (i.e. cutoffs/constraints)
+        atom_indices : list
+            Atom indicies of the model.
+        """
+        if ncmc:
+            #Defines ncmc move eqns for lambda peturbation of sterics/electrostatics
+            functions = { 'lambda_sterics' : 'min(1, (1/0.3)*abs(lambda-0.5))',
```

Comment:
As noted, it still bothers me I can't understand this with the info in the comments.